src/inference/src/dev/itensor.cpp

```diff
@@ -0,0 +1,490 @@
+        const size_t elem_size = get_element_type().size();
+        Strides byte_strides;
+        byte_strides.resize(element_strides.size());
+        std::transform(element_strides.begin(),
+                       element_strides.end(),
+                       byte_strides.begin(),
+                       [&elem_size](size_t stride) {
+                           return stride * elem_size;
+                       });
+        return byte_strides;
+    }
+
+    size_t get_size() const override {
+        return blob->size();
+    }
+
+    size_t get_byte_size() const override {
+        return blob->byteSize();
+    }
+
+    void* data(const element::Type& element_type) const override {
+        OPENVINO_ASSERT(blob != nullptr, "Tensor was not initialized.");
+#define TYPE_CHECK(TYPE) (dynamic_cast<const ie::TBlob<TYPE>*>(blob.get()) != nullptr)
+        auto host_accesable_implementation = TYPE_CHECK(bool) || TYPE_CHECK(int8_t) || TYPE_CHECK(uint8_t) ||
+                                             TYPE_CHECK(int16_t) || TYPE_CHECK(uint16_t) || TYPE_CHECK(int32_t) ||
+                                             TYPE_CHECK(uint32_t) || TYPE_CHECK(int64_t) || TYPE_CHECK(uint64_t) ||
+                                             TYPE_CHECK(float) || TYPE_CHECK(double);
+#undef TYPE_CHECK
+        OPENVINO_ASSERT(host_accesable_implementation,
+                        "Tensor implementation type dose not contains host accessable data");
+        if (element_type != element::undefined) {
+            OPENVINO_ASSERT(element_type == get_element_type(),
+                            "Tensor data with element type ",
+                            get_element_type(),
+                            ", is not representable as pointer to ",
+                            element_type);
+        }
+        // since we don't use byte offsets, we need to explicitly multiply by element_size
+        auto byte_offset = blob->getTensorDesc().getBlockingDesc().getOffsetPadding() * get_element_type().size();
+        OPENVINO_ASSERT((get_element_type().bitwidth() >= 8) || (byte_offset == 0),
+                        "ROI access for types with bitwidths less then 8 bit is not implemented. Tensor type: ",
+                        get_element_type());
+        return byte_offset + InferenceEngine::as<InferenceEngine::MemoryBlob>(blob)->rmap().as<uint8_t*>();
+    }
+
+    AnyMap get_properties() const override {
+        auto remote_impl = dynamic_cast<InferenceEngine::RemoteBlob*>(blob.get());
+        if (remote_impl != nullptr) {
+            auto params = remote_impl->getParams();
+            params.insert(ov::device::id(remote_impl->getDeviceName()));
+            return params;
+        } else {
+            return {};
+        }
+    }
+
+    std::shared_ptr<ie::Blob> blob;
+};
+
+class TensorRemoteBlob : public ie::RemoteBlob {
+public:
+    TensorRemoteBlob(const std::shared_ptr<ITensor>& tensor_)
```

Comment:
Is it an empty/fake blob to compatible with legacy RemoteBlob? We can see many of its method return null.

Reply:
It is a specific of InferenceEngine Remte blob